prow-configurations/prow-secrets-management.md

```diff
@@ -0,0 +1,76 @@
+# Prow secrets management
+
+## Overview
+
+Some jobs require using sensitive data. You need to encrypt data using KMS and store them in GCS.
+This document shows the commands necessary to create a service account and store its encrypted key in GCS bucket. This document assumes that you are logged in to the Google Cloud project with administrative rights.
+
+## Prerequisites
+
+ - [gcloud](https://cloud.google.com/sdk/gcloud/) 
+ - basic knowledge about [GCP key rings and keys](https://cloud.google.com/kms/docs/creating-keys).
+
+For your convenience, export these data:
+ - PROJECT_NAME - Google Cloud Project
+ - BUCKET_NAME - GCS bucket in the $PROJECT_NAME where Prow secrets are stored
+ - KEYRING_NAME - KMS key ring
+ - ENCRYPTION_KEY_NAME - key in the $KEYRING_NAME, used for data encryption
+
+## Secrets management
+
+>**NOTE:** Before following this guide, check Prow secrets setup for the $PROJECT_NAME.
+
+Set context to the $PROJECT_NAME executing:
+```
+gcloud config set project $PROJECT_NAME
+```
+
+### Create GCS bucket
+
+The purpose of the bucket is to store encrypted credentials necessary for prow jobs like provisioning clusters/vms etc.
+Run this command:
+```
+gsutil mb -p $PROJECT_NAME gs://$BUCKET_NAME/
+```
+
+### Create Google Service Account
+
+Export these variables:
+ - SA_NAME - Service Account name
+ - SA_DISPLAY_NAME - Service Account display name
+ - SECRET_FILE - the path where the private key will be written
+ - ROLE - the role bound to the $SA_NAME
+
+Create a service account:
+```
+gcloud iam service-accounts create $SA_NAME --display-name $SA_DISPLAY_NAME
+```
+
+Create a private key for the $SA_NAME service account:
+```
+gcloud iam service-accounts keys create $SECRET_FILE --iam-account=SA_NAME
+```
+
+Add policy binding for the $SA_NAME service account:
+```
+gcloud iam service-accounts add-iam-policy-binding $SA_NAME --member=serviceAccount:$SA_NAME@$PROJECT_NAME.iam.gserviceaccount.com --role=$ROLE
+```
+
+### Encrypt the secret
+
+Export the following:
```

Comment:
Export the **SECRET_FILE** variable which is the path to the file containing the secret.